test/e2e/autotls/domain_mapping_test.go

```diff
@@ -140,3 +155,145 @@ func TestDomainMappingAutoTLS(t *testing.T) {
+	names := test.ResourceNames{
+		Service: test.ObjectNameForTest(t),
+		Image:   test.PizzaPlanet1,
+	}
+
+	test.EnsureTearDown(t, clients, &names)
+
+	// service to be mapped
+	ksvc, err := v1test.CreateServiceReady(t, clients, &names)
+	if err != nil {
+		t.Fatalf("Failed to create initial Service %v: %v", names.Service, err)
+	}
+
+	host := ksvc.Service.Name + ".example.org"
+	// set resolvabledomain for custom domain to false by default.
+	resolvableCustomDomain := false
+
+	if test.ServingFlags.CustomDomain != "" {
+		host = ksvc.Service.Name + "." + test.ServingFlags.CustomDomain
+		resolvableCustomDomain = true
+	}
+
+	cert, key := makeCertificateForDomain(t, host)
+	secret, err := clients.KubeClient.CoreV1().Secrets(test.ServingNamespace).Create(context.Background(), &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: test.AppendRandomString("imasecret"),
+		},
+		Type: corev1.SecretTypeTLS,
+		Data: map[string][]byte{
+			corev1.TLSCertKey:       cert,
+			corev1.TLSPrivateKeyKey: key,
+		},
+	}, metav1.CreateOptions{})
+
+	if err != nil {
+		t.Fatalf("Domainmapping creation could not be completed: %v", err)
+	}
+
+	t.Cleanup(func() {
+		err = clients.KubeClient.CoreV1().Secrets(secret.Namespace).Delete(context.Background(), secret.Name, metav1.DeleteOptions{})
+		if err != nil {
+			t.Fatalf("Secret %s/%s could not be deleted: %v", secret.Namespace, secret.Name, err)
+		}
+	})
+
+	dm := v1alpha1.DomainMapping{
+		TypeMeta: metav1.TypeMeta{},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      host,
+			Namespace: ksvc.Service.Namespace,
+		},
+		Spec: v1alpha1.DomainMappingSpec{
+			Ref: duckv1.KReference{
+				APIVersion: "serving.knative.dev/v1",
+				Name:       ksvc.Service.Name,
+				Namespace:  ksvc.Service.Namespace,
+				Kind:       "Service",
+			}, TLS: &v1alpha1.SecretTLS{
+				SecretName:      secret.Name,
+				SecretNamespace: secret.Namespace,
+			}},
+		Status: v1alpha1.DomainMappingStatus{},
+	}
+
+	t.Log("HOST===>", host)
+	_, err = clients.ServingAlphaClient.DomainMappings.Create(context.Background(), &dm, metav1.CreateOptions{})
+	if err != nil {
+		t.Fatalf("Problem creating domainmapping : %v", err)
+	}
+
+	if waitErr := wait.PollImmediate(test.PollInterval, test.PollTimeout, func() (bool, error) {
+		ingresses, err := clients.NetworkingClient.Ingresses.Get(context.Background(), ksvc.Service.Name, metav1.GetOptions{})
+		if err != nil {
+			return false, err
+		}
+		for _, cond := range ingresses.Status.Conditions {
+			if cond.Type == apis.ConditionReady {
+				return cond.IsTrue(), nil
+			}
+		}
+		return false, nil
+	}); waitErr != nil {
+		t.Fatalf("Ingresses %s never became ready: %v", ksvc.Service.Name, waitErr)
+	}
+
+	err = shared.CheckDistribution(context.Background(), t, clients,
+		&url.URL{
+			Host:   host,
+			Scheme: "HTTPS",
+		}, 1, 1, []string{test.PizzaPlanetText1}, resolvableCustomDomain)
+	if err != nil {
+		t.Fatalf("Hitting %v didn't return the expected response: %v", dm.Name, err)
+	}
```

Comment:
The `Kingress` is sorta an implementation detail of the `DomainMapping` - we should be checking the `Ready` condition on the `DomainMapping` itself

I also wonder if we `CheckDistribution` isn't needed and maybe we could just use `WaitForEndpointState`